process/block/metrics.go

```diff
@@ -161,42 +117,43 @@ func saveMetricsForCommitMetachainBlock(
 		epoch = epoch - 1
 	}
 
+	incrementMetricCountConsensusAcceptedBlocks(header, epoch, nodesCoordinator, appStatusHandler, managedPeersHolder)
+}
+
+func incrementMetricCountConsensusAcceptedBlocks(
+	header data.HeaderHandler,
+	epoch uint32,
+	nodesCoordinator nodesCoordinator.NodesCoordinator,
+	appStatusHandler core.AppStatusHandler,
+	managedPeersHolder common.ManagedPeersHolder,
+) {
 	pubKeys, err := nodesCoordinator.GetConsensusValidatorsPublicKeys(
-		header.PrevRandSeed,
-		header.Round,
+		header.GetPrevRandSeed(),
+		header.GetRound(),
 		core.MetachainShardId,
```

Comment:
is this ok? Shouldn't have been header.ShardID() ?

Reply:
right, messed it up while merging the methods